executable-spec/src/Cardano/Ledger/Spec/STS/Update.hs

```diff
@@ -38,15 +52,22 @@ data UPDATE hashAlgo
 -- adding more components to this environment.
 data Env hashAlgo
   = Env
-    { ideationEnv :: Environment (IDEATION hashAlgo)
+    { currentSlot :: !Slot
+    , closedVotingPeriods :: !(Map (SIPHash hashAlgo) (VotingPeriod hashAlgo))
```

Comment:
Open and closed voting periods should be part of the state I think, since they are related to the update system.